### ISSUE TITLE
Upload cleanup on finish

### ISSUE THREAD

**Issue by olegmoz** [artipie/docker-adapter#137]
Upload data should be deleted when blob has been uploaded. Client does not send DELETE request on finish, so it should be done in the end of PUT request handling.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by 0crat** [comment]
I'm not managing `artipie/docker-adapter` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by g4s8** [comment]
@olenagerasimova please check this ticket

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@66d967ea-fa4a-451b-bad8-e94a595ef078, thread: pool-770-thread-1
com.zerocracy.radars.github.ThrottledComments[74] java.lang.IllegalStateException: Can't post an identical message to artipie/docker-adapter#137: I'm not managing `artipie/docker-adapter` GitHub repository, you have to contact me in Slack first, 
```

[0.53.14](https://github.com/zerocracy/farm): Issue: artipie/docker-adapter#137, Action: `assigned`

**Comment by olenagerasimova** [comment]
@olegmoz do I understand correctly, that `Upload` needs `delete` method, that have to implemented by deleting content by `AstoUpload#data` key? 
Also I have questions about `UploadEntity.Put#response` implementation: 
1) should not we first compare digests and only then, if they match, put content to `blobStore`? 
2) should upload data be removed if digests do not match?

**Comment by olegmoz** [comment]
@olenagerasimova when an upload is deleted everything that is stored by it's root key should be removed. Right now it is only `data` file if anything. But that may change in future, we need to store multiple chunks for upload in future along with some meta info.

1. That is true, it would be better to compare digests before uploading blob content, not after as it is done now.
2. I cannot see a direct answer for this in docs. It would be nice to set an experiment with real docker registry and see what happens in this case.

**Comment by 0pdd** [comment]
@olegmoz 2 puzzles [#142](https://github.com/artipie/docker-adapter/issues/142), [#143](https://github.com/artipie/docker-adapter/issues/143) are still not solved.

**Comment by olenagerasimova** [comment]
@olegmoz could this issue be closed now?

**Comment by olegmoz** [comment]
@olenagerasimova yes, closing it now. Would you like to take on related puzzles?

**Comment by 0crat** [comment]
I'm not managing `artipie/docker-adapter` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by olenagerasimova** [comment]
@olegmoz yes, assign them to me please

**Comment by 0pdd** [comment]
@olegmoz 3 puzzles [#143](https://github.com/artipie/docker-adapter/issues/143), [#157](https://github.com/artipie/docker-adapter/issues/157), [#158](https://github.com/artipie/docker-adapter/issues/158) are still not solved; solved: [#142](https://github.com/artipie/docker-adapter/issues/142).

**Comment by 0pdd** [comment]
@olegmoz 3 puzzles [#143](https://github.com/artipie/docker-adapter/issues/143), [#157](https://github.com/artipie/docker-adapter/issues/157), [#175](https://github.com/artipie/docker-adapter/issues/175) are still not solved; solved: [#142](https://github.com/artipie/docker-adapter/issues/142), [#158](https://github.com/artipie/docker-adapter/issues/158).

**Comment by 0pdd** [comment]
@olegmoz 2 puzzles [#143](https://github.com/artipie/docker-adapter/issues/143), [#157](https://github.com/artipie/docker-adapter/issues/157) are still not solved; solved: [#142](https://github.com/artipie/docker-adapter/issues/142), [#158](https://github.com/artipie/docker-adapter/issues/158), [#175](https://github.com/artipie/docker-adapter/issues/175).

**Comment by 0pdd** [comment]
@olegmoz the puzzle [#143](https://github.com/artipie/docker-adapter/issues/143) is still not solved; solved: [#142](https://github.com/artipie/docker-adapter/issues/142), [#157](https://github.com/artipie/docker-adapter/issues/157), [#158](https://github.com/artipie/docker-adapter/issues/158), [#175](https://github.com/artipie/docker-adapter/issues/175).

**Comment by 0pdd** [comment]
@olegmoz all 5 puzzles are solved here: [#142](https://github.com/artipie/docker-adapter/issues/142), [#143](https://github.com/artipie/docker-adapter/issues/143), [#157](https://github.com/artipie/docker-adapter/issues/157), [#158](https://github.com/artipie/docker-adapter/issues/158), [#175](https://github.com/artipie/docker-adapter/issues/175).
